src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.LocalFunctionReferenceRewriter.cs

```diff
@@ -0,0 +1,146 @@
+        /// </summary>
+        private sealed class LocalFunctionReferenceRewriter : BoundTreeRewriterWithStackGuard
+        {
+            private readonly LambdaRewriter _lambdaRewriter;
+
+            public LocalFunctionReferenceRewriter(LambdaRewriter lambdaRewriter)
+            {
+                _lambdaRewriter = lambdaRewriter;
+            }
+
+            public override BoundNode VisitCall(BoundCall node)
+            {
+                if (node.Method.MethodKind == MethodKind.LocalFunction)
+                {
+                    BoundExpression receiver;
+                    MethodSymbol method;
+                    var arguments = node.Arguments;
+                    _lambdaRewriter.RemapLocalFunction(node.Syntax, node.Method, out receiver, out method, ref arguments);
+                    node = node.Update(receiver, method, arguments);
+                }
+
+                var visited = base.VisitCall(node);
+                if (visited.Kind != BoundKind.Call)
+                {
+                    return visited;
+                }
+
+                return (BoundCall)visited;
+            }
+
+            public override BoundNode VisitDelegateCreationExpression(BoundDelegateCreationExpression node)
+            {
+                if (node.MethodOpt?.MethodKind == MethodKind.LocalFunction)
+                {
+                    BoundExpression receiver;
+                    MethodSymbol method;
+                    var arguments = default(ImmutableArray<BoundExpression>);
+                    _lambdaRewriter.RemapLocalFunction(
+                        node.Syntax, node.MethodOpt, out receiver, out method, ref arguments);
+
+                    return new BoundDelegateCreationExpression(
+                        node.Syntax, receiver, method, isExtensionMethod: false, type: node.Type);
+                }
+
+                return base.VisitDelegateCreationExpression(node);
+            }
+
+            public override BoundNode VisitConversion(BoundConversion conversion)
+            {
+                if (conversion.ConversionKind == ConversionKind.MethodGroup &&
+                    conversion.SymbolOpt?.MethodKind == MethodKind.LocalFunction)
+                {
+                    BoundExpression receiver;
+                    MethodSymbol method;
+                    var arguments = default(ImmutableArray<BoundExpression>);
+                    _lambdaRewriter.RemapLocalFunction(
+                        conversion.Syntax, conversion.SymbolOpt, out receiver, out method, ref arguments);
+
+                    return new BoundDelegateCreationExpression(
+                        conversion.Syntax, receiver, method, isExtensionMethod: false, type: conversion.Type);
+                }
+                return base.VisitConversion(conversion);
+            }
+
+            public override BoundNode VisitBlock(BoundBlock node)
+            {
+                var newStatements = ArrayBuilder<BoundStatement>.GetInstance();
```

Comment:
That's a good point. I will add a test for this and make sure the behavior is correct.